mne/viz/utils.py

```diff
@@ -897,3 +901,126 @@ def _process_times(inst, times, n_peaks=None, few=False):
+        Info structure containing the channel locations.
+    kind : str
+        Whether to plot the sensors as 3d or as topomap. Available options
+        'topomap', '3d'. Defaults to 'topomap'.
+    ch_type : 'mag' | 'grad' | 'eeg' | 'seeg' | None
+        The channel type to plot. If None, then channels are chosen in the
+        order given above.
+    title : str | None
+        Title for the figure. If None (default), equals to
+        ``'Sensor positions (%s)' % ch_type``.
+    show_names : bool
+        Whether to display all channel names. Defaults to False.
+    show : bool
+        Show figure if True. Defaults to True.
+
+    Returns
+    -------
+    fig : instance of matplotlib figure
+        Figure containing the sensor topography.
+
+    Notes
+    -----
+    This function plots the sensor locations from the info structure using
+    matplotlib. For drawing the sensors using mayavi see
+    :func:`mne.viz.plot_trans`.
+    See also :func:`mne.viz.plot_layout`.
+
+    .. versionadded:: 0.12.0
+
+    """
+    if kind not in ['topomap', '3d']:
+        raise ValueError("Kind must be 'topomap' or '3d'.")
+    ch_indices = channel_indices_by_type(info)
+    if ch_type is None:
+        for type in ['mag', 'grad', 'eeg', 'seeg']:
+            if _contains_ch_type(info, type):
+                ch_type = type
+                break
+    elif ch_type not in ['mag', 'grad', 'eeg', 'seeg']:
+        raise ValueError("ch_type must be one of ['mag', 'grad', 'eeg', "
+                         "'seeg'] not %s!" % ch_type)
+    picks = ch_indices[ch_type]
+    if kind == 'topomap':
+        pos = _auto_topomap_coords(info, picks, True)
+    else:
+        pos = np.asarray([ch['loc'][:3] for ch in info['chs']])[picks]
+    def_colors = _handle_default('color')
+    ch_names = np.array(info['ch_names'])[picks]
+    bads = [idx for idx, name in enumerate(ch_names) if name in info['bads']]
+    colors = ['red' if i in bads else def_colors[channel_type(info, pick)]
+              for i, pick in enumerate(picks)]
+    title = 'Sensor positions (%s)' % ch_type if title is None else title
+    fig = _plot_sensors(pos, colors, ch_names, title, show_names, show)
+
+    return fig
+
+
+def _plot_sensors(pos, colors, ch_names, title, show_names, show):
+    """Helper function for plotting sensors."""
+    import matplotlib.pyplot as plt
+    from mpl_toolkits.mplot3d import Axes3D
+    from .topomap import _check_outlines
+    fig = plt.figure()
+
+    def onpick(event, pos, ch_names):
+        ind = event.ind[0]  # Just take the first sensor.
+        ch_name = ch_names[ind]
+        this_pos = pos[ind]
+
+        # XXX: Bug in matplotlib won't allow setting the position of existing
+        # text item, so we create a new one.
+        ax.texts.pop(0)
+        if len(this_pos) == 3:
+            ax.text(this_pos[0], this_pos[1], this_pos[2], ch_name)
+        else:
+            ax.text(this_pos[0], this_pos[1], ch_name)
+        fig.canvas.draw()
```

Comment:
did we have a policy about nested functions? if there is no strong reason for keeping it here make it private and tear it out.